File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceCaseDecisionUpdateRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceCaseDecisionUpdateRequest {

    @NotNull
    private final Long documentID;

    @NotNull
    private final Date updatedDateOfFill;

    @NotNull
    private final Time updatedTimeOfFill;

    @NotNull
    @NotEmpty
    private final String updatedPlaceOfFill;

    @NotNull
    private final Long updatedCulpritID;

    @NotNull
    @NotEmpty
    private final String updatedCulpritActualPlaceOfResidence;

    @NotNull
    @NotEmpty
    private final String updatedCaseCircumstances;

    @NotNull
    @NotEmpty
    private final String updatedCaseDecision;

    @NotNull
    private final Date updatedDateOfReceiving;

    @NotNull
    private final Date updatedEffectiveDate;

    @NotNull
    private final Long carAccidentEntityID;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceCaseDecisionUpdateRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceCaseDecisionUpdateRequest(
    val documentID: Long,
    val updatedDateOfFill: Date,
    val updatedTimeOfFill: Time,
    val updatedPlaceOfFill: String,
    val updatedCulpritID: Long,
    val updatedCulpritActualPlaceOfResidence: String,
    val updatedCaseCircumstances: String,
    val updatedCaseDecision: String,
    val updatedDateOfReceiving: Date,
    val updatedEffectiveDate: Date,
    val carAccidentEntityID: Long
)
